### ISSUE TITLE
squoosh-cli no longer converts from png to avif on Ubuntu or MacOS since yesterday ( June 16 2021 )

### ISSUE THREAD

**Issue by datacorruption** [GoogleChromeLabs/squoosh#1057]
**Description**
squoosh-cli no longer converts from png to avif on Ubuntu or MacOS since yesterday (June 16, 2021)

`squoosh-cli --resize '{"enabled":true,"width":706,"height":1442,"method":"lanczos3","fitMethod":"stretch","premultiply":true,"linearRGB":true}' --avif '{"cqLevel":33,"cqAlphaLevel":-1,"denoiseLevel":0,"tileColsLog2":0,"tileRowsLog2":0,"speed":6,"subsample":1,"chromaDeltaQ":false,"sharpness":0,"tune":0}' 0001.png

⠋ (node:21354) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 17 unpipe listeners added to [WriteStream]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
(node:21354) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 17 error listeners added to [WriteStream]. Use emitter.setMaxListeners() to increase limit
(node:21354) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 17 close listeners added to [WriteStream]. Use emitter.setMaxListeners() to increase limit
(node:21354) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 17 finish listeners added to [WriteStream]. Use emitter.setMaxListeners() to increase limit
0/1 ⠙ ▐╌╌╌╌╌╌╌╌╌╌▌ Decoding...
node:internal/process/promises:227
          triggerUncaughtException(err, true /* fromPromise */);
          ^
[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "`unwrap_throw` failed".] {
  code: 'ERR_UNHANDLED_REJECTION'
}`

**To Reproduce**
install squoosh on Mac/Ubuntu
`npm install -g @squoosh/cli`
Convert png to avif ( command copied from squoosh.app )
`squoosh-cli --resize '{"enabled":true,"width":706,"height":1442,"method":"lanczos3","fitMethod":"stretch","premultiply":true,"linearRGB":true}' --avif '{"cqLevel":33,"cqAlphaLevel":-1,"denoiseLevel":0,"tileColsLog2":0,"tileRowsLog2":0,"speed":6,"subsample":1,"chromaDeltaQ":false,"sharpness":0,"tune":0}' 0001.png`

**Expected behavior**
It should convert to avif. And it was! Until very recently

**Version:**
 - OS w/ version: Ubuntu 20.04 LTS / MacOS Catalina 10.15.4
 - Browser w/ version: N/A (cli issue)
 - Node version: 15.8.0
 - npm version: 7.5.4

**Is your issue related to the quality of image compression?**
No

**Comment by mjpieters** [comment]
I can reproduce this by using https://www.fnordware.com/superpng/pngtest16rgba.png as the input; presumably it's the alpha channel that is causing issues?

**Comment by ChucKN0risK** [comment]
I have the same issue.

**Comment by surma** [comment]
@mjpieters Your image seems to be 16bit. This should have been fixed in the most recent release of the Squoosh CLI.

@datacorruption Does your problem still occur with the most recent release of the Squoosh CLI?

@ChucKN0risK Can you provide a sample image?

**Comment by ChucKN0risK** [comment]
Sample image
![2021_07_29-sources-light](https://user-images.githubusercontent.com/3913276/127469527-aaa72150-de93-4b08-9856-08a304098b55.jpg)

Bug screenshot
![CleanShot 2021-07-29 at 11 38 57](https://user-images.githubusercontent.com/3913276/127469553-971105dc-8c80-45e6-b2f3-bc5bf27c1b5e.jpg)

**Comment by surma** [comment]
@ChucKN0risK Thank you! I think your CLI invocation might be slightly incorrect.
Can you try again and replace `--webp --mozjpeg` with `--webp '{}' --mozjpeg '{}'? If that fixes it, then it’s “just“ a matter of parameter parsing that needs fixing.

**Comment by datacorruption** [comment]
> @datacorruption Does your problem still occur with the most recent release of the Squoosh CLI?

@surma Updated to the latest version of the squoosh cli and re-ran the command from my initial post on the exact same file that was originally causing this issue. Can confirm that it is now working!

Looking through this thread, after seeing the comment from @mjpieters and your response, I checked and it appears that the file that was giving us a problem was also a 16-bit png.

Also tested with the sample image posted by @ChucKN0risK and confirmed to be working. Suggest opening a different issue related to parameter parsing if necessary.

Resolved in squoosh-cli v7.20.3 -- closing this issue.
